src/js/stores/BallotStore.js

```diff
@@ -110,14 +110,20 @@ class BallotStore extends FluxMapStore {
     let newBallot = {};
 
     switch (action.type) {
+      // Dale 2016-06-02 On some pages we are calling BallotActions.retrieve() as many as three times in order to
+      // avoid problems like Issue #224. This should be optimized.
 
       case "voterRetrieve":
+        // console.log("BallotStore, voterRetrieve");
+        // We need to refresh the ballot here (in addition to voterAddressRetrieve) because there are routes that you
+        // can go to directly (e.g. wevote.me/ballot) that don't reach out and retrieve the ballot items any other way.
         BallotActions.retrieve();
```

Comment:
I did test it without the BallotActions.retrieve() under "voterRetrieve", and I found a couple of cases where no ballot was returned. I am sure that after this milestone we can review the retrieve dependencies and optimize the code to only make the voterBallotItemsRetrieve call once. I am glad to have a solution that solves the problem from the perspective of the voter, even though it requires extra calls to the same API endpoint in certain circumstances.